vividus-plugin-accessibility/src/main/java/org/vividus/accessibility/AccessibilitySteps.java

```diff
@@ -126,14 +131,30 @@ public void performAccessibilityScan(ExamplesTable checkOptions)
                 publishAttachment(axeOptionsAsString, currentUrl, Map.of("entries", reportEntries, "url", currentUrl,
                         "run", axeOptions));
 
-                long failures = reportEntries.stream().filter(e -> ResultType.FAILED == e.getType())
-                        .map(AxeReportEntry::getResults).map(List::size).findFirst().orElse(0);
-                softAssert.assertThat(String.format(AXE_CORE_MESSAGE, axeOptionsAsString, currentUrl), failures,
-                        equalTo(0L));
+                List<Result> failures = reportEntries.stream().filter(e -> ResultType.FAILED == e.getType())
+                        .map(AxeReportEntry::getResults).findFirst().orElse(List.of());
+                reportViolations(failures, Result::getId, Result::getDescription, AXE_CORE_MESSAGE, axeOptionsAsString,
+                        currentUrl);
             });
         }
     }
 
+    private <T> void reportViolations(List<T> violations, Function<T, String> idGetter,
+            Function<T, String> messageGetter, String format, Object... args)
+    {
+        if (reportViolationsAsAssertions)
+        {
+            violations.stream().map(v -> "[" + idGetter.apply(v) + "] " + messageGetter.apply(v))
+                    .forEach(softAssert::recordFailedAssertion);
+
+            if (!violations.isEmpty())
+            {
+                return;
+            }
+        }
+        softAssert.assertThat(String.format(format, args), (long) violations.size(), equalTo(0L));
```

Comment:
```suggestion
        }
        else
        {
            softAssert.assertThat(String.format(format, args), (long) violations.size(), equalTo(0L));
        }
```

Reply:
fixed, but it was intentional to perform this soft assert just to keep step non empty when there are no violations

Reply:
merge pls :)